pkg/reconciler/v1alpha1/taskrun/resources/output_resource.go

```diff
@@ -0,0 +1,188 @@
+/*
+Copyright 2018 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1"
+	listers "github.com/knative/build-pipeline/pkg/client/listers/pipeline/v1alpha1"
+	buildv1alpha1 "github.com/knative/build/pkg/apis/build/v1alpha1"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	outputDir = "/workspace/output/"
+)
+
+// AddOutputResources reads the output resources and adds the corresponding container steps
+// This function also reads the inputs to check if resources are redeclared in inputs and has any custom
+// target directory.
```

Comment:
why can't we require that the output path will always be in `output/resource_name` even if its different in the input, this way it at least will be consistent and easier for users to know what to expect, it might require copying it to the output folder if its an output

Reply:
You mean if user declares output resources then expect the resource to be present in `output/resource_name` folder?

Then user has to add a step to copy the resource to the folder path `/output/resource_name`. I think it might be too much work. I agree its more predictable but I was trying to infer the path from already given information which means less yaml. I do not have strong opinion about this as both methods as have pros and cons.

Reply:
If resource is defined in input then in code we can figure out where is the resource placed. I do not see value in copying from input resource destination -> /output/resource_name -> external bucket. As of current implementation we have input resource destination -> external bucket
1. This would introduce additional copy step. 
2. code complexity is not solved but just moved to input resource. Now input resource has to check if resource is defined in output and add some additional step to ensure copy has happened before resolving outputs. I am not sure of this change